WdtBase.cpp

```diff
@@ -59,10 +71,49 @@ ErrorCode WdtUri::process(const string& url) {
     paramsIndex = urlPiece.size();
   }
   ErrorCode status = OK;
-  hostName_.assign(urlPiece.data(), paramsIndex);
-  if (hostName_.size() == 0) {
+  StringPiece hostNamePiece(urlPiece, 0, paramsIndex);
+  if (hostNamePiece.empty()) {
     LOG(ERROR) << "URL doesn't have a valid host name " << url;
     status = URI_PARSE_ERROR;
+  } else {
+    bool isIpv6 = false;
+    size_t addrEndIndex = hostNamePiece.find("]");
```

Comment:
Check directly hostNamePiece[end-1] why search also start checking the first character
(Add test with malformed/and short like "[", "]" ...)